src/Api/ShareaholicApi.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Drupal\shareaholic\Api;
+
+use Drupal;
+use Drupal\Core\Config\ImmutableConfig;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\RequestOptions;
+use Psr\Log\LoggerInterface;
+
+class ShareaholicApi {
+  const SERVICE_URL = 'https://www.shareaholic.com';
+  const API_URL = 'https://web.shareaholic.com';
+  const HEALTH_CHECK_URL = self::API_URL . '/haproxy_health_check';
+  const KEY_GENERATING_URL = self::API_URL . '/publisher_tools/anonymous';
+  const EVENTS_URL = self::API_URL . '/api/events';
+  const SESSIONS_URL = self::API_URL . '/api/v3/sessions';
+
+  /** @var Client */
+  private $httpClient;
+
+  /** @var ImmutableConfig */
+  private $config;
+
+  /** @var LoggerInterface */
+  private $logger;
+
+  public function __construct(Client $httpClient, ImmutableConfig $config, LoggerInterface $logger) {
+    $this->httpClient = $httpClient;
+    $this->config = $config;
+    $this->logger = $logger;
+  }
+
+  /**
+   * @return string|null
+   */
+  public function generateApiKey($verificationKey, $siteName, $langcode) {
+
+    $post_data = [
+      'configuration_publisher' => [
+        'verification_key' => $verificationKey,
+        'site_name' => $siteName,
+        'domain' => \Drupal::request()->getHost(),
+        'platform_id' => '2',
+        'language_id' => $this->getLanguageId($langcode),
+        'shortener' => 'shrlc',
+        'recommendations_attributes' => [
+          'locations_attributes' => [],
```

Comment:
Location lists need to be updated (with enable/disable status) to send the list of locations to create when the API is generated.